kura/test/org.eclipse.kura.core.crypto.test/src/main/java/org/eclipse/kura/core/crypto/EncryptDecryptTest.java

```diff
@@ -0,0 +1,135 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ ******************************************************************************/
+package org.eclipse.kura.core.crypto;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.io.IOUtils;
+import org.eclipse.kura.KuraException;
+import org.junit.Before;
+import org.junit.Test;
+
+public class EncryptDecryptTest {
+
+    private CryptoServiceImpl cryptoService;
+
+    @Before
+    public void setup() {
+        this.cryptoService = new CryptoServiceImpl();
+    }
+
+    @Test
+    public void shouldStreamEncodeAndStreamDecodeCorrectly() throws KuraException, IOException {
+
+        String dataToEncrypt = "testingcrypto";
+        File tempFile = new File("/tmp/testingCrypto.txt");
+        tempFile.deleteOnExit();
+
+        try (FileOutputStream out = new FileOutputStream(tempFile);
```

Comment:
There is no need to write to a file for testing this, you can write to a ByteArrayOutputStream and then read from the result of its getBytes() method using a ByteArrayInputStream